### DISCUSSION TITLE
Start contributing to your project.

### DISCUSSION THREAD

**Discussion by akanakis** [OpenTracksApp/OpenTracks#700] [General]
Hello,
I'm a student, I started using your application and I find it very interesting, so I would  like to contribute to it. Could you please suggest me some issues or anything else to start with, because I'm beginner at contributing?
Thank you for your time.

**Comment by dennisguse** [comment]
Really cool!

Some things that are somehow cool to work on (depending on your preference)
* GPX enhancement: https://github.com/OpenTracksApp/OpenTracks/issues/652
  (should be not too complicated) )
* rendering an image, so people can share it: https://github.com/OpenTracksApp/OpenTracks/issues/363
* enable automation for OpenTracks https://github.com/OpenTracksApp/OpenTracks/issues/671
  (mainly Android magic required)

PS: Test coverage is also an issue in the code base.
It is (still) rather old and there some, but some things are hard to test with the current implementation.
So, if you build something new, it is good too add tests as those also document the functionality and help to maintain the system.

**Reply by akanakis** [reply]
Ok, thank you very much. I will try my best!

**Reply by dennisguse** [reply]
For easier coordination:
*  just add a comment to the issue you want to work on.
* open one PR per issue (keep changes reviewable, so we can merge them quickly)
* and if you need some more info or something is missing, don't hesitate to ask.
  It is often easier/nicer to handle this before somebody invested a lot of work.
* ff you find something that looks can be improved (code wise), just open a new issue/PR.

Anything else I can help you with?

PS: Our tests are currently failing in Github Actions, so don't be upset if the tests fails in your PR.
It is a memory issue in the CI's emulator.

**Reply by akanakis** [reply]
Ok, thank you very much again! I will follow your advice and if I need help, I will ask you immediately.
